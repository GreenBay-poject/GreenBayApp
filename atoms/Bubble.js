import { StatusBar } from 'expo-status-bar';
import React from 'react';
import { StyleSheet, Text, TouchableHighlight, View } from 'react-native';
import { AntDesign } from '@expo/vector-icons';
import { DARK_GREEN, LIGHT_SILVER } from '../shared/colors';

export default function Bubble({ by, ans, title, question, date }) {

  console.log(title.split('@').slice(1).join())
  var title_updated = title.split('@').slice(1).join()
  return (
    <View>
      <View style={{ marginVertical: 12, width: '80%', alignSelf: 'flex-start', backgroundColor: LIGHT_SILVER, padding: 12, borderRadius: 13 }}>
        <Text style={{ fontSize: 13, fontWeight: 'bold', marginBottom: 10 }}>{title_updated}</Text>
        <Text style={{ fontSize: 12 }}>{question}</Text>
        <Text style={{ alignSelf: 'flex-end', fontSize: 10 }}>asked by {by}</Text>
        <Text style={{ alignSelf: 'flex-end', fontSize: 10 }}>on {date}</Text>
      </View>
      {
        ans == null ?
          <View style={{ marginVertical: 12, alignSelf: 'flex-end', backgroundColor: LIGHT_SILVER, padding: 12, borderRadius: 13 }}>
            <Text style={{ fontSize: 13, alignSelf: 'flex-start', fontWeight: 'bold', marginBottom: 10 }}>Not Answered Yet</Text>
          </View>
          : <View style={{ marginVertical: 12, width: '80%', alignSelf: 'flex-end', backgroundColor: LIGHT_SILVER, padding: 12, borderRadius: 13 }}>

            <Text style={{ fontSize: 12, alignSelf: 'flex-start' }}>{ans[0]['Answere']}</Text>
            <Text style={{ alignSelf: 'flex-end', fontSize: 10 }}>answered by {ans[0]['AuthorsID']}</Text>
            <Text style={{ alignSelf: 'flex-end', fontSize: 10 }}>on {ans[0]['DatePosted']}</Text>
          </View>
      }

    </View>

  );
}

const styles = StyleSheet.create({

});
