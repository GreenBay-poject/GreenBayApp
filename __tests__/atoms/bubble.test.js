import React from 'react';
import renderer from 'react-test-renderer';

import Bubble from '../../atoms/Bubble'

describe('<Bubble />', () => {
    it('With Null Answer', () => {
        const tree = renderer.create(<Bubble by={"Saman Perera"} ans={null} title={"Sample Title"} question={"Sample Question"} date={"2021-11-02"} />).toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree.children.length).toBe(2);
        expect(tree.children[1].children[0].children[0]).toBe('Not Answered Yet');
        //console.log(tree.children[1].children[0].children[0])
    });
    it('With Not Null Answer', () => {
        const tree = renderer.create(<Bubble by={"Saman Perera"} ans={"Sample Answer"} title={"Sample Title"} question={"Sample Question"} date={"2021-11-02"} />).toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree.children.length).toBe(2);
        expect(tree.children[1].children.length).toBe(3);
        //console.log(tree.children[1].children.length).toBe(3)
    });
});