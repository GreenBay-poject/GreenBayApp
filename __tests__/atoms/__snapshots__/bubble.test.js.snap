// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bubble /> With Not Null Answer 1`] = `
<View>
  <View
    style={
      Object {
        "alignSelf": "flex-start",
        "backgroundColor": "#d7dbd8",
        "borderRadius": 13,
        "marginVertical": 12,
        "padding": 12,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 13,
          "fontWeight": "bold",
          "marginBottom": 10,
        }
      }
    >
      
    </Text>
    <Text
      style={
        Object {
          "fontSize": 12,
        }
      }
    >
      Sample Question
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "flex-end",
          "fontSize": 10,
        }
      }
    >
      asked by 
      Saman Perera
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "flex-end",
          "fontSize": 10,
        }
      }
    >
      on 
      2021-11-02
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "flex-end",
        "backgroundColor": "#d7dbd8",
        "borderRadius": 13,
        "marginVertical": 12,
        "padding": 12,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "flex-start",
          "fontSize": 12,
        }
      }
    />
    <Text
      style={
        Object {
          "alignSelf": "flex-end",
          "fontSize": 10,
        }
      }
    >
      answered by 
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "flex-end",
          "fontSize": 10,
        }
      }
    >
      on 
    </Text>
  </View>
</View>
`;

exports[`<Bubble /> With Null Answer 1`] = `
<View>
  <View
    style={
      Object {
        "alignSelf": "flex-start",
        "backgroundColor": "#d7dbd8",
        "borderRadius": 13,
        "marginVertical": 12,
        "padding": 12,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 13,
          "fontWeight": "bold",
          "marginBottom": 10,
        }
      }
    >
      
    </Text>
    <Text
      style={
        Object {
          "fontSize": 12,
        }
      }
    >
      Sample Question
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "flex-end",
          "fontSize": 10,
        }
      }
    >
      asked by 
      Saman Perera
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "flex-end",
          "fontSize": 10,
        }
      }
    >
      on 
      2021-11-02
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "flex-end",
        "backgroundColor": "#d7dbd8",
        "borderRadius": 13,
        "marginVertical": 12,
        "padding": 12,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "flex-start",
          "fontSize": 13,
          "fontWeight": "bold",
          "marginBottom": 10,
        }
      }
    >
      Not Answered Yet
    </Text>
  </View>
</View>
`;
